cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -316,28 +321,68 @@ struct dispatch_t<StableAddress,
       return config.error();
     }
 
-    auto loaded_bytes_per_iter = kernel_source.LoadedBytesPerIteration();
-    // choose items per thread to reach minimum bytes in flight
-    const int items_per_thread =
-      loaded_bytes_per_iter == 0
-        ? +policy.ItemsPerThreadNoInput()
-        : ::cuda::ceil_div(policy.min_bif, config->max_occupancy * block_dim * loaded_bytes_per_iter);
-
-    // but also generate enough blocks for full occupancy to optimize small problem sizes, e.g., 2^16 or 2^20 elements
-    const int items_per_thread_evenly_spread = static_cast<int>(
-      (::cuda::std::min)(Offset{items_per_thread}, num_items / (config->sm_count * block_dim * config->max_occupancy)));
+    auto can_vectorize = false;
+    // the policy already handles the compile-time checks if we can vectorize. Do the remaining alignment check here
+#ifdef CCCL_C_EXPERIMENTAL
+    if (Algorithm::vectorized == wrapped_policy.GetAlgorithm())
+#else // CCCL_C_EXPERIMENTAL
+    if constexpr (Algorithm::vectorized == ActivePolicy::algorithm)
+#endif // CCCL_C_EXPERIMENTAL
+    {
+      const int alignment     = wrapped_policy.LoadStoreWordSize();
+      auto is_pointer_aligned = [&](auto it) {
+        if constexpr (THRUST_NS_QUALIFIER::is_contiguous_iterator_v<decltype(it)>)
+        {
+          return THRUST_NS_QUALIFIER::detail::util::is_aligned(
+            THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator(it), alignment);
+        }
+        else
+        {
+          return true; // fancy iterators are aligned, since the vectorized kernel chooses a different code path
+        }
+      };
+      can_vectorize = (is_pointer_aligned(::cuda::std::get<Is>(in)) && ...) && is_pointer_aligned(out);
+    }
 
-    const int items_per_thread_clamped =
-      ::cuda::std::clamp(items_per_thread_evenly_spread, +policy.MinItemsPerThread(), +policy.MaxItemsPerThread());
-    const int tile_size = block_dim * items_per_thread_clamped;
+    const int ipt = [&] {
+#ifdef CCCL_C_EXPERIMENTAL
+      if (Algorithm::vectorized == wrapped_policy.GetAlgorithm())
+#else // CCCL_C_EXPERIMENTAL
+      if constexpr (Algorithm::vectorized == ActivePolicy::algorithm)
+#endif // CCCL_C_EXPERIMENTAL
```

Comment:
I still believe this should rather be 
```suggestion
      if _CCCL_CONSTEXPR_IF_NOT_PYTHON_COMPILATION (Algorithm::vectorized == wrapped_policy.GetAlgorithm())
```

Reply:
I am rolling this back, unfortunately :/